src/hotspot/share/cds/lambdaProxyClassDictionary.hpp

```diff
@@ -160,6 +161,10 @@ class DumpTimeLambdaProxyClassDictionary
                              137, // prime number
                              ResourceObj::C_HEAP> {
 public:
+  // GCC will memset for default ctor. With user defined ctor, data members need manually initialized.
+  // We could not use default ctor, it will destroy allocation_type in debug version and fail to delete.
+  // see bug 8269537
+  DumpTimeLambdaProxyClassDictionary() : _count(0)  {}
```

Comment:
Since you changed the construction call from 
```
(ResourceObj::C_HEAP, mtClass)DumpTimeLambdaProxyClassDictionary();
```
to
```
(ResourceObj::C_HEAP, mtClass)DumpTimeLambdaProxyClassDictionary;
```

I think it's no longer necessary to add the `RunTimeLambdaProxyClassInfo()` and `DumpTimeSharedClassTable()` constructors in this PR. In 8269537, we should find a generic way to prevent this bug from happening.

Reply:
Initialization of _count is still needed. I could remove DumpTimeSharedClassTable, but still, we should keep the ctor which initialize the two data members --- though they will be reset in update_counts. I will remove the comments.